code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -4,6 +4,8 @@
 	var/obj/item/organ/eyes/E = getorganslot(ORGAN_SLOT_EYES)
 	if(!E)
 		return INFINITY //Can't get flashed without eyes
+	if(!(~sight & (SEE_MOBS|SEE_OBJS|SEE_TURFS)) && !(istype(E, /obj/item/organ/eyes/robotic/xray/syndicate) || istype(E, /obj/item/organ/eyes/robotic/shield)))
+		return 0
```

Comment:
Poor code quality, should be a var that protects it and checking the sight vars is pretty dodgy imo

Reply:
you can't get those 3 from anything other than x-ray

Reply:
That doesn't mean they could be added in the future and that it's not dodgy code